File: first-quarter/introduction-to-programming/keywords-and-operators/keywords-and-operators.ts
// This file is an example of how to use keywords and operators in TypeScript.

// This line is required to use the global scope.
// The export keyword is used to make the file a module.
export {}

// The declare keyword is used to tell TypeScript that the following code already exists.
// In this case, we're telling TypeScript that the global scope has a String interface.
declare global {
    interface String {
        /**
         * Returns the last character of a string.
         * @example
         * const str = 'Hello, world!'
         * console.log(str.length) // expected output: 13
         */
        ending: string
    }
}

// We'll assume that the String interface has been implemented in the global scope.
// In TypeScript specifically, this needs to be done by the programmer.
const my_string: String = 'Hello, world!'
console.log(my_string.ending) // expected to be '!' but is undefined as of now

File: first-quarter/introduction-to-programming/keywords-and-operators/keywords-and-operators.cpp
// This file is an example of how keywords and operators work in C++.

// The include keyword is used to include the contents of another file in this file.
#include <bits/stdc++.h>

// The int keyword is used to declare an instance of the int class.
int x = 10;

// The main keyword is used to declare the main function.
// The int keyword before the main keyword is used to declare the return type of the main function.
// The () after the main keyword is used to declare the parameters of the main function.
// The {} after the () is used to declare the scope of the main function.
int main() {
    // The = operator is used to assign the value 20 to the variable x.
    int x = 20;

    {
        int x = 30;

        // The std::cout keyword is used to access the cout object from the std namespace.
        // The << operator is used to insert the value of x into the cout object.
        // The std::endl keyword is used to insert a newline character into the cout object.
        std::cout << x << std::endl; // 30
    }

    std::cout << x << std::endl; // 20

    // The return keyword is used to return the value 0 from the main function.
    return 0;
}
